src/internal/_Set.js

```diff
@@ -0,0 +1,167 @@
+var _contains = require('./_contains');
+
+
+// A simple Set type that honours R.equals semantics
+module.exports = (function() {
+  function _Set() {
+    /* globals Set */
+    this._nativeSet = typeof Set === 'undefined' ? null : new Set();
+    this._items = {};
+    this._negZeroExists = false;
+    this._nullExists = false;
+    this._undefExists = false;
+    this._falseExists = false;
+    this._trueExists = false;
```

Comment:
Not necessary for native Sets. The same would have to exist for environments that don't support native Sets though, so I figured we could reuse it to prevent an extra lookup in the native Set.